File: src/types/tool.ts
export interface ToolStatus {
  installed: boolean;
  path: string | null;
  version: string | null;
  install_method: string | null;
  running: boolean;
}

export interface InstallResult {
  success: boolean;
  message: string;
  output: string | null;
}

export interface PlatformInfo {
  platform: string;
  platform_name: string;
  arch: string;
}

export type ToolId = "claude-code" | "codex";

File: src-tauri/src/models/tool.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct ToolStatus {
    pub installed: bool,
    pub path: Option<String>,
    pub version: Option<String>,
    pub install_method: Option<String>,
    pub running: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct InstallResult {
    pub success: bool,
    pub message: String,
    pub output: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct PlatformInfo {
    pub platform: String,
    pub platform_name: String,
    pub arch: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct NodeStatus {
    pub node_installed: bool,
    pub node_version: Option<String>,
    pub nvm_installed: bool,
    pub nvm_version: Option<String>,
}
